partitioned-table.md

```diff
@@ -987,43 +991,90 @@ When reorganizing partitions, you need to note the following key points:
     ERROR 1526 (HY000): Table has no partition for value 6
     ```
 
-### Manage Hash partitions
+- After reorganize partition, the statistics needs to be updated
 
-Unlike Range partitioning, `DROP PARTITION` is not supported in Hash partitioning.
+```sql
++---------+------+--------------------------------------------------------------------------------------------------------------------------------------------------------+
+| Level   | Code | Message                                                                                                                                                |
++---------+------+--------------------------------------------------------------------------------------------------------------------------------------------------------+
+| Warning | 1105 | The statistics of related partitions will be outdated after reorganizing partitions. Please use 'ANALYZE TABLE' statement if you want to update it now |
++---------+------+--------------------------------------------------------------------------------------------------------------------------------------------------------+
+1 row in set (0.00 sec)
+```
 
-Currently, `ALTER TABLE ... COALESCE PARTITION` is not supported in TiDB as well. For partition management statements that are not currently supported, TiDB returns an error.
+### Manage Hash and Key partitions
 
-{{< copyable "sql" >}}
+Example table:
 
 ```sql
-ALTER TABLE MEMBERS OPTIMIZE PARTITION p0;
+CREATE TABLE example (id int primary key, data varchar(1024)) PARTITION BY HASH(id) PARTITIONS 2;
 ```
 
+Increase the number of partitions by 1:
+
+```sql
+ALTER TABLE example ADD PARTITION PARTITIONS 1;
+```
+
+One can also specify partition options by adding partitions by partition definitions:
+
 ```sql
-ERROR 8200 (HY000): Unsupported optimize partition
+ALTER TABLE example ADD PARTITION
+(PARTITION pExample4 COMMENT = 'not p3, but pExample4 instead',
+ PARTITION pExample5 COMMENT = 'not p4, but pExample5 instead');
 ```
 
-### Key partition management
+Unlike Range/List partitioning, `DROP PARTITION` is not supported in Hash/Key partitioning, but you can decrease the number of partitions with `COALESCE PARTITION` or delete all data from specific partitions with `TRUNCATE PARTITION`
 
-Currently, Key partitioning only supports the `ALTER TABLE ... TRUNCATE PARTITION` partition management statement.
+Decreasing number of partitions by 1:
 
 ```sql
-ALTER TABLE members TRUNCATE PARTITION p0;
+ALTER TABLE example COALESCE PARTITION 1;
 ```
 
+Notes:
+- Changing the number of partitions for Hash/Key partitioned tables will do an online reorganization/copy all data to the new number of partitions.
+- After changing the number of partitions, the statistics needs to be updated
+
```

Comment:
question: Should users manually update the statistics for that table or TiDB will do that automatically after the reorganization?

Reply:
Unfortunately we have not yet had time to fix the statistics update, so it updates the statistics automatically during or after ALTER TABLE, so users needs to update the statistics for that table after the DDL.

A bit more details is that it will actually keep the 'global statistics' which is used for tidb_partition_prune_mode = 'Dynamic', but I hope this will be more user friendly in the future.